backend/main.go

```diff
@@ -42,20 +45,61 @@ const (
 type Server struct {
 	store        storage.StorageService
 	storeTimeout time.Duration
+	auth         Auth
+}
+
+type Auth interface {
+	authenticate(next http.HandlerFunc) http.HandlerFunc
+}
+
+type noAuth struct{}
+
+func (auth noAuth) authenticate(next http.HandlerFunc) http.HandlerFunc {
+	return next
+}
+
+type firebaseAuth struct {
+	firebaseClient authClient
+}
+
+type authClient interface {
+	VerifyIDToken(ctx context.Context, idToken string) (*auth.Token, error)
+}
+
+func (auth firebaseAuth) authenticate(next http.HandlerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		ctx, cancel := context.WithTimeout(r.Context(), defaultStoreTimeout)
```

Comment:
We shouldn't use `defaultStoreTimeout` here. This was only intended for the storage (and can be overridden by the user anyway).

We should have a separate timeout on the `Server` struct, called `authTimeout` or something like that.

The value of `defaultStoreTimeout` may be an equally reasonable default for both the storage and the auth, in which case it would be fine to rename that constant.

Reply:
I've put a default auth timeout into the auth package.